src/Reflex/Main.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Reflex.Main where
+
+import Data.Dependent.Sum (DSum (..) )
+import Control.Concurrent (Chan, readChan, forkIO, ThreadId)
+import Reflex.Host.Class
+import Reflex.PerformEvent.Base
+import Reflex.PostBuild.Base
+import Reflex.TriggerEvent.Base
+import Data.IORef (readIORef)
+import Control.Monad.Ref (Ref, MonadRef)
+import Data.Maybe (catMaybes)
+import Control.Monad.IO.Class
+import Control.Monad (forever)
+import Data.Traversable (for)
+import Data.Foldable (for_)
+import Data.Functor.Identity
+
+type EventChannel t = Chan [DSum (EventTriggerRef t) TriggerInvocation]
+
+runHost
```

Comment:
Let's give this a slightly longer name, like `runHostWithIO`, to indicate that it grants access to IO.  I don't want people to get the impression that all hosts can/should support IO.